cmd/soroban-rpc/internal/methods/get_events.go

```diff
@@ -510,6 +512,7 @@ func (h eventsRPCHandler) getEvents(ctx context.Context, request GetEventsReques
 	return GetEventsResponse{
 		LatestLedger: ledgerRange.LastLedger.Sequence,
 		Events:       results,
+		EndLedger:    endLedger,
```

Comment:
it seems possible that `EndLedger ` is greater than `LatestLedger` which is confusing because someone could mistakenly think that all ledgers up to `EndLedger` were scanned. However, we will never scan any ledgers beyond `LatestLedger`

Reply:
This can be fixed by making sure endLedger does not exceed the retention_window so `min(endLedger, LatestLedger)`